src/bson.ts

```diff
@@ -21,38 +32,26 @@ export {
   BSONRegExp,
   BSONSymbol,
   Map,
-  deserialize,
-  serialize,
-  calculateObjectSize
+  Document
 } from 'bson';
 
-/** @public */
-export interface Document {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  [key: string]: any;
-}
+import type { DeserializeOptions, SerializeOptions } from 'bson';
 
-import type { SerializeOptions } from 'bson';
-
-// TODO: Remove me when types from BSON are updated
 /**
  * BSON Serialization options.
  * @public
  */
-export interface BSONSerializeOptions extends Omit<SerializeOptions, 'index'> {
-  /** Return document results as raw BSON buffers */
-  fieldsAsRaw?: { [key: string]: boolean };
-  /** Promotes BSON values to native types where possible, set to false to only receive wrapper types */
-  promoteValues?: boolean;
-  /** Promotes Binary BSON values to native Node Buffers */
-  promoteBuffers?: boolean;
-  /** Promotes long values to number if they fit inside the 53 bits resolution */
-  promoteLongs?: boolean;
-  /** Serialize functions on any object */
-  serializeFunctions?: boolean;
-  /** Specify if the BSON serializer should ignore undefined fields */
-  ignoreUndefined?: boolean;
-
+export interface BSONSerializeOptions
+  extends Omit<SerializeOptions, 'index'>,
```

Comment:
we discovered that Serialize and Deserialize options end up being effectively the same here, so this code can be simplified accordingly

Reply:
I looked over this too quickly in our discussion, there is a difference and these keys below are the ones we desire from the underlying BSON library:

serialize options: `checkKeys` `serializeFunctions` `ignoreUndefined`
deserialize options: `promoteLongs` `promoteBuffers` `promoteValues` `fieldsAsRaw`

I added a typescript test to cover this expectation.